docs/en/data_types/special_data_types/interval.md

```diff
@@ -0,0 +1,74 @@
+# Interval {#data-type-interval}
+
+Represents time and date intervals. The resulting type of the `INTERVAL` operator.
+
+!!!warning "Warning"
+    You can't use the `Interval` data type for storing values in tables.
+
+Structure:
+
+- Unsigned integer value of interval.
+- Type of interval.
+
+Supported interval types:
+
+- SECOND
+- MINUTE
+- HOUR
+- DAY
+- WEEK
+- MONTH
+- QUARTER
+- YEAR
+
+For each interval type, there is the separated data type. For example, the DAY interval is expressed as the `IntervalDay` data type:
+
+```sql
+SELECT toTypeName(INTERVAL 4 DAY)
+```
+```text
+┌─toTypeName(toIntervalDay(4))─┐
+│ IntervalDay                  │
+└──────────────────────────────┘
+```
+
+## Usage Remarks {#data-type-interval-usage-remarks}
+
+You can use `Interval`-type values in arithmetical operations with [Date](../../data_types/date.md) and [DateTime](../../data_types/datetime.md)-type values. For example, you can add 4 days to the current time:
+
+```sql
+SELECT now() as current_date_time, current_date_time + INTERVAL 4 DAY
+```
+```text
+┌───current_date_time─┬─plus(now(), toIntervalDay(4))─┐
+│ 2019-10-23 10:58:45 │           2019-10-27 10:58:45 │
+└─────────────────────┴───────────────────────────────┘
+```
+
+Intervals of different types can't be combined in one `INTERVAL` operator. You can't use the expressions like `INTERVAL 4 DAY 1 HOUR`, so you need to express intervals in the minimal units. For example, `1 day and an hour` interval should be expressed as `INTERVAL 25 HOUR`.
```

Comment:
* this is important and should be explained earlier, probably in !!! block, but not in terms of INTERVAL operator, but in terms of Interval values
* minimal units = SECOND, it's more like "smallest unit it consists of"